SIGUSR1 for rehash the SSL certificates

drscream opened the pull request.

drscream committed:
SIGUSR1 for rehash the SSL certificates

drscream committed:
nice comments please

wiedi commented:
this will leak `ctx` and probably also the error strings from `SSL_load_error_strings()`

drscream committed:
ssl_init: fix ctx leak and one-time global init on rehash

Free the previous SSL_CTX before replacing it so repeated SIGUSR1
rehashes do not leak it. Guard SSLeay_add_ssl_algorithms() and
SSL_load_error_strings() with a static flag so they only run once.
